src/viam/components/board/client.py

```diff
@@ -48,12 +48,14 @@ async def read(
         extra: Optional[Dict[str, Any]] = None,
         timeout: Optional[float] = None,
         **__,
-    ) -> int:
+    ) -> Board.Analog.Value:
         if extra is None:
             extra = {}
         request = ReadAnalogReaderRequest(board_name=self.board.name, analog_reader_name=self.name, extra=dict_to_struct(extra))
         response: ReadAnalogReaderResponse = await self.board.client.ReadAnalogReader(request, timeout=timeout)
-        return response.value
+        result = self.Value(value=response.value, min_range=response.min_range,
+                            max_range=response.max_range, step_size=response.step_size)
+        return result
```

Comment:
you should just be able to do `return response` now

Reply:
Good idea! Take one more look.